kratos/tests/utilities/test_geometrical_projection_utilities.cpp

```diff
@@ -0,0 +1,106 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:     BSD License
+//  			 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+//
+
+// Project includes
+#include "testing/testing.h"
+#include "utilities/geometrical_projection_utilities.h"
+
+#include "geometries/triangle_3d_3.h"
+
+namespace Kratos
+{
+namespace Testing
+{
+
+using NodeType = typename GeometricalProjectionUtilities::NodeType;
+using GeometryType = typename GeometricalProjectionUtilities::GeometryType;
+
+GeometryType::Pointer CreateTriangle2D3NForTest()
+{
+    GeometryType::PointsArrayType points;
+    points.push_back(Kratos::make_shared<NodeType>(0.04, 0.02, 0.0));
+    points.push_back(Kratos::make_shared<NodeType>(1.1, 0.03, 0.0));
+    points.push_back(Kratos::make_shared<NodeType>(1.08, 1.0, 0.0));
+
+    return GeometryType::Pointer(new Triangle3D3<NodeType>(points));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeometricalProjectionUtilitiesFastProjectDirection, KratosCoreFastSuite)
+{
+    GeometryType::Pointer p_geom = CreateTriangle2D3NForTest();
+
+    double expected_proj_dist = 1.258;
+
+    const double x_coord = 0.325;
+    const double y_coord = 0.147;
+
+    Point point_to_proj(x_coord, y_coord, expected_proj_dist);
+
+    array_1d<double,3> dir_vector;
+    array_1d<double,3> normal_vector;
+
+    dir_vector[0] = 0.0;
```

Comment:
~~~cpp
    dir_vector[0] = 0.0;
    dir_vector[0] = 0.0;
    dir_vector[0] = 1.0;

    normal_vector[0] = 0.0;
    normal_vector[0] = 0.0;
    normal_vector[0] = -1.0;
~~~

I found the error, one hint, thee is something wrong in the indexes

Reply:
Sorry for this ...